gulpfile.js

```diff
@@ -82,6 +83,68 @@ function createReloadServer() {
   });
 }
 
+function writeServiceWorkerFile() {
+  let path = require('path');
+  let rootDir = 'dist';
+
+  let config = {
+    cacheId: 'polymerproject',
+    staticFileGlobs: [
+      `${rootDir}/1.0/index.html`,
+      `${rootDir}/1.0/shell.html`,
+      `${rootDir}/1.0/about.html`,
+      `${rootDir}/elements/**`,
+      `${rootDir}/js/*.js`,
+      `${rootDir}/css/*.css`,
+      `${rootDir}/bower_components/**/webcomponents-lite.min.js`,
+    ],
+    dynamicUrlToDependencies: {
+      '/1.0/shell.html': [
+        path.join(rootDir, 'templates', 'base-devguide.html'),
+        path.join(rootDir, 'templates', 'head-meta.html'),
+        path.join(rootDir, 'templates', 'site-nav.html'),
+        path.join(rootDir, 'templates', 'base-blog.html'),
+        path.join(rootDir, 'templates', 'page-md-styles.html'),
+      ]
+    },
+    navigateFallback: '/1.0/shell.html',
+    runtimeCaching: [
+    {
+      urlPattern: /\/images\/*\//,
+      handler: 'fastest',
+    },
+    {
+      urlPattern: /\/1\.0\/docs\//,
+      handler: 'fastest',
+    },
+    {
+      urlPattern: /\/1\.0\/start\//,
+      handler: 'fastest',
+    },
+    {
+      urlPattern: /\/1\.0\/toolbox\//,
+      handler: 'fastest',
+    },
+    {
+      urlPattern: /\/1\.0\/homepage\//,
+      handler: 'fastest',
+    },
+    {
+      urlPattern: /\/1\.0\/blog\//,
+      handler: 'fastest',
+      options: {
+        cache: {
+          maxEntries: 10,
+          name: 'blog-cache'
+        }
+      }
+    }],
+    stripPrefix: rootDir + '/',
+    verbose: false  /* When debugging, you can enable this to true  */
+  };
+  swPrecache.write(path.join(rootDir, 'service-worker.js'), config);
```

Comment:
This should be prefixed with a `return` so that the Promise is propagated back to the `gulp.task()`.